Cargo.toml

```diff
@@ -36,9 +36,14 @@ rayon = "1.9"
 serde = "1.0"
 serde_derive = "1.0, <=1.0.171"  # avoid precompiled binaries (https://github.com/serde-rs/serde/issues/2538)
 serde_json = "1.0"
-signal-hook = "0.3"
 tiny_http = { version = "0.12", optional = true }
 
+[target.'cfg(windows)'.dependencies]
+ctrlc = "3.2.5"
```

Comment:
Just from curiosity, why not using 3.4.2 `ctrlc`?
IIUC it should support our MSRV: https://github.com/Detegr/rust-ctrlc/blob/3.4.2/Cargo.toml#L15